### PR TITLE
Use correct CLR types at GetMostPreciseType

### LINKED ISSUE #102: CompareUsingMostPreciseType fails on Int32
If a==null and b.GetType() is Int32, the comparison fails because it is not recognized as a basic type. Only Int64 is included in that set.

### PR DESCRIPTION
**Pull request by gumbarros**
This closes #102

### CODE REVIEW

File: src/NCalc/Domain/EvaluationVisitor.cs [hunk 1]
```diff
@@ -699,11 +715,8 @@ public override void Visit(Identifier parameter)
         if (Parameters.ContainsKey(parameter.Name))
```
Comment (Bykiev) [inline src/NCalc/Domain/EvaluationVisitor.cs:715]: you can replace `ContainsKey` with `TryGetValue` and then use the retrieved value on line 718 and 729

Comment (yallie) [reply src/NCalc/Domain/EvaluationVisitor.cs:715]: Good point! Thanks

**Comment by Beckdotan** [comment]
can you please return CompareUsingMostPreciseType to be public? I'm using it since i wrote a new visit function in this code:
https://dotnetfiddle.net/5iSl8A
Since I wanted to create my own logic for null values.

**Comment by gumbarros** [comment]
@Beckdotan sorry for the unintended change, I published 3.13.1 with the fix

**Pull request merged**